content/docs/user-guide/dvc-metafile-formats.md

```diff
@@ -0,0 +1,128 @@
+  default
+
+Note that comments can be added to DVC metafiles using the `# comment` syntax.
+
+> `.dvc` file comments are preserved among executions of the `dvc repro` and
+> `dvc commit` commands, but not when a `.dvc` file is overwritten by
+> `dvc add`,`dvc import`, or `dvc import-url`.
+
+## dvc.yaml file
+
+When you add commands to a pipeline with `dvc run`, the `dvc.yaml` file is
+created or updated. Here's a simple example:
+
+```yaml
+stages:
+  firstone:
+    cmd: python cmd.py input.data output.data metrics.json
+    deps:
+    - cmd.py
+    - input.data
+    outs:
+    - output.data
+    metrics:
+    - metrics.json
+  nextone:
+    cmd: python ...
+    ...
+```
+
+`dvc.yaml` files consists of a group of `stages` with names provided explicitly
+by the user with the `--name` (`-n`) option of `dvc run`. Each stage can contain
+the following fields:
+
+- `cmd` - executable command defined in this stage
+- `deps` - list of <abbr>dependencies</abbr> for this stage
+- `params` - (optional) list of the [parameter](/doc/command-reference/params)
+  names and their current values
+- `outs` - list of <abbr>outputs</abbr> for this stage
+- `metric` - (optional) list of [metric](/doc/command-reference/metrics) files
+- `locked` - (optional) whether or not this stage is locked from reproduction
+- `always_changed` (optional) - whether or not this stage is considered as
+  changed by commands such as `dvc status` and `dvc repro`. `false` by default
+
+A dependency entry consists of a these possible fields:
+
+- `path`: Path to the dependency, relative to the `wdir` path (always present)
+- `md5`: MD5 hash for the dependency (most [stages](/doc/command-reference/run))
+- `etag`: Strong ETag response header (only HTTP <abbr>external
+  dependencies</abbr> created with `dvc import-url`)
+- `repo`: This entry is only for external dependencies created with
+  `dvc import`, and can contains the following fields:
+
+  - `url`: URL of Git repository with source DVC project
+  - `rev`: Only present when the `--rev` option of `dvc import` is used.
+    Specific commit hash, branch or tag name, etc. (a
+    [Git revision](https://git-scm.com/docs/revisions)) used to import the
+    dependency from.
+  - `rev_lock`: Git commit hash of the external <abbr>DVC repository</abbr> at
+    the time of importing or updating (with `dvc update`) the dependency.
+
+  > See the examples in
+  > [External Dependencies](/doc/user-guide/external-dependencies) for more
+  > info.
+
+An output entry consists of these fields:
+
+- `md5` - hash value for the output file
+- `path` - path to the output in the <abbr>workspace</abbr>, relative to the
+  location of the `.dvc` file
+- `cache` - (optional) whether or not DVC should cache the output. `true` by
+  default
+
+Metrics entries can contain these fields:
+
+- `type`: Type of the metric file (`json`)
+- `xpath`: Path within the metric file to the metrics data(e.g. `AUC.value` for
+  `{"AUC": {"value": 0.624321}}`)
```

Comment:
I'm not sure yet about all these details. Probably some things are missing and others have been deprecated.

Reply:
Has been reviewed by Ruslan in https://github.com/iterative/dvc.org/pull/1370#pullrequestreview-426091399.